mesonbuild/backend/backends.py

```diff
@@ -1000,13 +999,18 @@ def get_target_link_deps_mappings(self, t, prefix):
         mapping of the install_name value to the new one, so we can change them
         on install.
         '''
+        dylib_cache = {}
         result = {}
         if isinstance(t, build.StaticLibrary):
             return result
         for ld in t.get_all_link_deps():
             if ld is t or not isinstance(ld, build.SharedLibrary):
                 continue
-            old = get_macos_dylib_install_name(ld.prefix, ld.name, ld.suffix, ld.soversion)
+            if ld in dylib_cache:
+                old = dylib_cache[ld]
+            else:
+                old = get_macos_dylib_install_name(ld.prefix, ld.name, ld.suffix, ld.soversion)
+                dylib_cache[ld] = old
```

Comment:
A simpler way to cache this would be to add `@functools.lru_cache()` as a decorator on `get_macos_dylib_install_name`.

Reply:
I don't understand why, but this only saves up to 2 sec. while this cache here safes up to 6 sec.... :( 
Therefore i kept the dict here.

Reply:
Could you add a comment about the difference in performance so that people don't accidentally revert it in the future?

Reply:
Actually, the difference might come from the `maxsize` param. Can you try setting it to `None`? And also for the other usage. Having a lot of targets probably makes the difference.

Reply:
Sure :) added!

Reply:
I will check

Reply:
Okay, it turned out that the cache seems to work correctly if i only cache the ld and not over the 4 parameters. Is this also okay for you this way ?